### PR TITLE
[TRANSLATED] pt_BR 1-5 (please someone review"

### PR DESCRIPTION
**Pull request by grstavares**

### CODE REVIEW

**Comment by twostraws** [comment]
@lmbcosta Can you help?

**Comment by lmbcosta** [comment]
Yup. I will review

File: pt-BR/1-5.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+Esta propriedade será uma string - o nome da imagem a ser carregada - mas ela precisa ser uma string *opcional* porque quando o view controller for criado inicialmente ele não saberá o valor da propriedade. Logo após a criação ela será preenchida, mas ainda assim, quando o view controller for inicializado ela estará vazia.
+
+Então, adicione esta propriedade ao `DetailViewController`, logo após o `@IBOutlet`:
+
+    var selectedImage: String?
+
+A primeira tarefa foi realizada, vamos para a próxima: implementar `didSelectRowAt` de maneira a carregar o `DetailViewController` a partir do storyboard.
+
+Quando nós criamos o view controller de detalhes, você preencheu o storyboardID "Detail", o que nos permite carregar ele a partir do storyboard usando um método com o nome `instantiateViewController(withIdentifier:)`. Todos view controllers possuem uma propriedade chamada `storyboard` que faz referência ao storyboard pelo qual o view controller foi instanciado, ou nil. No caso do `ViewController` esta referência será ao Main.storyboard, que é o mesmo storyboard que contém o detail view controller. Por isto, vamos usar esta referência para chamá-lo.
```
Comment (lmbcosta) [inline pt-BR/1-5.md]: typo: Todos os view controllers instead Todos view controllers. Not sure

Comment (grstavares) [reply pt-BR/1-5.md]: I think you're right!

File: pt-BR/1-5.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+Segundo, mesmo tendo certeza que o método `instantiateViewController()` irá nos responder com `DetailViewController`, Swift *acredita* que ele irá nos responder com um `UIViewController` porque a linguagem não consegue interpretar qual o tipo exato que foi descrito dentro do storyboard.
+
+Isto parecerá confuso se você é novo em programação, por isso, deixe-me tentar te explicar melhor usando uma analogia. Vamos dizer que você tem um encontro hoje a noite, diante disto você me pede para lhe conseguir dois ingressos para um evento. Sendo seu amigo eu saio e procuro os ingressos, coloco-os em um envelope e lhe entrego. Eu realizei a minha parte do combinado: Você pediu os ingressos, recebeu os ingressos. Mas que tipo de ingressos são? Ingressos para o teatro? Para um jogo de esportes? Para o metrô? A única maneira de você saber é abrir o envelope e conferir.
```
Comment (lmbcosta) [inline pt-BR/1-5.md]: typo: hoje à noite instead hoje a noite. 
I'm sorry, I'm Portuguese and sometimes I 'm not sure about pt-BR grammar.
Please ignore comments if not make sense to you.

File: pt-BR/1-5.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+O termo técnico para este procedimento em programação é "typecasting", o que se fosse traduzido para o português seria "conversão de tipo". Com este procedimento nós dizemos para qualquer linguagem de programação que se baseie em tipos para tratar determinado valor como sendo de um tipo diferente do tipo que ela espera. Com Swift temos diversas maneiras de fazer isto, por agora iremos usar a opção mais segura. E a segurança que ela nos proporciona é porque realizamos o nosso typecasting usando opcionais. O que faremos, se fosse traduzido para o português, seria mais ou menos o seguinte: "Tente converter o tipo que você recebrá para este outro tipo e caso seja bem sucedido na conversão use o valor. Caso não seja bem sucedido, não faça nada e siga em frente."
```
Comment (lmbcosta) [inline pt-BR/1-5.md]: typo: converter o tipo que você receberá instead converter o tipo que você recebrá

Comment (grstavares) [reply pt-BR/1-5.md]: Right again!!!

File: pt-BR/1-5.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+Se conseguirmos obter um `UIViewController` convertido para um `DetailViewController` iremos definir a sua propriedade `selectedImage` com o valor que existe em `picures[indexPath.row]`, da mesma maneira que fizemos em `cellForRowAt` - Segunda mini-tarefa realizada!
+
+A terceira mini-tarefa é fazer a nova tela aparecer. Você já viu que view controllers possuem uma propriedade opcional chamada `storyboard` que pode conter o storyboard que os inicializou ou estar nula em caso de ele não ter sido inicializado a partir de um storyboard. Da mesma forma existe uma outra propriedade nos view controllers com o nome de `navigationController` que conterá o navigation controller em que ele está inserido caso este seja o caso, ou nulo em caso contrário.
```
Comment (lmbcosta) [inline pt-BR/1-5.md]: typo: Você já viu que os view controllers instead Você já viu que view controllers

File: pt-BR/1-5.md (hunk 2 above)
Comment (grstavares) [reply pt-BR/1-5.md]: In that case I'm pretty sure that it does not apply

File: pt-BR/1-5.md (hunk 4 above)
Comment (grstavares) [reply pt-BR/1-5.md]: this one I think that accept the two alternatives. Maybe it became better with your suggestion. Please feel free to change.

Comment (lmbcosta) [inline pt-BR/1-5.md]: storyboard que os inicializou ou estar nula em caso de ele não ter sido inicializado a partir de um storyboard. 
Suggestion: I think you should use nil instead nulo or nula like you use english words: view controller, storyboard...

File: pt-BR/1-5.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        if let vc = storyboard?.instantiateViewController(withIdentifier: "Detail") as? DetailViewController {
+            // 2: Sucesso! Defina o valor da propriedade selectedImage
+            vc.selectedImage = pictures[indexPath.row]
+            
+            // 3: Agora, adicione o DetailViewController ao navigation controller
+            navigationController?.pushViewController(vc, animated: true)
+        }
+    }
+
+Vamos dar uma olhada na linha do `if let` com um pouco mais de detalhe por um momento. Existem três partes que podem falhar: a propriedade `storyboard` pode ser nil (neste caso o `?` irá impedir que o restante da linha seja executado), o `instantiateViewController()` pode falhar, caso tenhamos passado qualquer storyboard ID inexistent ("sbrobles", por exemplo). A conversão de tipo - `as?` - também pode falhar, se recebermos de volta um view controller que não é compatível com o tipo que indicamos.
```
Comment (lmbcosta) [inline pt-BR/1-5.md]: storyboard ID inexistente instead storyboard ID inexistent

File: pt-BR/1-5.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+    if let imageToLoad = selectedImage {
+        imageView.image  = UIImage(named: imageToLoad)
+    }
+
+A primeira linha checa e desembrulha o valor dentro da propriedae `selectedImage`. Se por alguma razão `selectedImage` é nulo (o que em teoria nunca vai acontecer), então a instrução `imageView.image` nunca será executada. Se por outro lado ela possuir um valor este será incluído na variável `imageToLoad` e passado como parametro para `UIImage` e então carregado.
+
+Ok, é isto: pressione play ou Cmd+R para executar a aplicação e testar o novo código. Você deve ser capaz de selecionar qualquer uma das figuras para que elas possam entrar deslizando para serem mostradas em tela cheia.
+
+Perceba que apareceu um botão Voltar na barra de navegação que permite ao usuário retornar ao `ViewController`. Se você clicar e arrastar com delicadeza, perceberá que consegue fazer a mesma operação usando gestos de deslizar - pressione o extremo esquerdo da tela e deslize para a direita, como você faria usando o telefone.
```
Comment (lmbcosta) [inline pt-BR/1-5.md]: Actually the user will see "Back" not "Voltar" on the navigation bar. You can translate if you want. But I think you should mention what the user will see.

File: pt-BR/1-5.md (hunk 2 above)
Comment (lmbcosta) [reply pt-BR/1-5.md]: Ignore comment

**Comment by lmbcosta** [comment]
Review done

File: pt-BR/1-5.md (hunk 4 above)
Comment (lmbcosta) [reply pt-BR/1-5.md]: Fine by me

**Comment by twostraws** [comment]
@grstavares Are you able to apply these changes so we can merge?

**Comment by grstavares** [comment]
Updated!

**Pull request merged**